confgenerator/testdata/valid/linux/all-built_in_config/golden_otel.conf

```diff
@@ -3,6 +3,8 @@ exporters:
     metric:
       instrumentation_library_labels: false
       prefix: ""
+      resource_filters:
+      - prefix: sqlserver.database.name
```

Comment:
As MSSQL metrics are windows only, i am trying to understand why the PR is touching linux side configuration files..

Reply:
@sophieyfang great question. Here is a link to a conversation about this in a previous PR https://github.com/GoogleCloudPlatform/ops-agent/pull/542/files#diff-bcf4db7b1ead9873183e1c4b42f5022443169df5866d48e713032a239e9d74f6

Reply:
That conversation seems to be answering a different question. Sophie's question is about whether we support the "mssql" receiver at all on Linux VMs. I vaguely recall there were some customers that uses Microsoft SQL Server on Linux. I dunno why but shrug...

Reply:
@qingling128 are you sure she's not referring to the conversation in that thread? we do no support the mssql receiver on Linux VMs. but the reason why that specific configuration has "resource_filters: - prefix: sqlserver.database.name" was because of the code that had to be added to the google exporter (which applies across all linux and windows receivers) if i recall correctly.

Reply:
OH I see. That made the connections. We are discussing internally for how to mitigate this impact. Right now there are some concerns that the logic unnecessarily applies too broadly to all metrics. But it is a limitation with the current features supported by the upstream.